### DISCUSSION TITLE
Choppy mouse on high refresh rates with G-Sync

### DISCUSSION THREAD

**Discussion by zsol** [wezterm/wezterm#4016] [General]
Given a Windows machine with a high resolution and high refresh rate display with G-Sync enabled, I saw the mouse cursor go choppy every time wezterm was focused.

I was hunting this issue for a few days until I realized: the screen refresh rate drastically drops (from 144 Hz) whenever the terminal is focused, which is what's causing the mouse choppiness. I suspect this is due to the combination of the following:

* G-Sync is enabled for windowed applications in my NVIDIA Control Panel: 
![image](https://github.com/wez/wezterm/assets/66740/cd30a2fe-7382-4b9f-8be3-a9ec7f3258ab)
* Due to the way wezterm renders itself, it's categorized as a "game" by nvidia, applying the G-Sync setting above
* but wezterm can't render at 144 Hz, so the refresh rate of the whole display is dropped to match the terminal's rendering performance

Not sure there's much point in optimizing wezterm's rendering pipeline for high refresh rates (which is why this isn't an issue in wezterm IMO), so this is the workaround I have:
![image](https://github.com/wez/wezterm/assets/66740/34589675-2620-4fcd-b4da-22231f017487)
In NVIDIA Control Panel, go to "Manage 3D Settings", click the "Program Settings" tab in the main pane, then find or add wezterm.exe as a program to customize. Make sure the "Monitor Technology" setting is set to "Fixed Refresh". Apply, and restart wezterm.

You could also disable G-Sync for windowed applications, I suppose.

**Comment by wez** [comment]
Thanks for sharing!
Have you experimented with different `front_end` settings to see whether those trigger this differently?
https://wezfurlong.org/wezterm/config/lua/config/front_end.html

**Comment by wez** [comment]
Also: you might want to try setting https://wezfurlong.org/wezterm/config/lua/config/max_fps.html to 144
wezterm almost certainly won't be able to process changes to the contents of the terminal at 144Hz, but things like updating mouse selection should be able to run quickly.
